ddtrace/internal/telemetry/metrics_namespaces.py

```diff
@@ -42,45 +40,36 @@ class MetricNamespace:
     }
 
     def __init__(self):
+        # type: () -> None
         self._flush()
 
     def _flush(self):
+        # type: () -> None
         self._metrics_data = {
-            TELEMETRY_TYPE_GENERATE_METRICS: {TELEMETRY_NAMESPACE_TAG_TRACER: {}, TELEMETRY_NAMESPACE_TAG_APPSEC: {}},
-            TELEMETRY_TYPE_DISTRIBUTION: {TELEMETRY_NAMESPACE_TAG_TRACER: {}, TELEMETRY_NAMESPACE_TAG_APPSEC: {}},
-        }
+            TELEMETRY_TYPE_GENERATE_METRICS: defaultdict(dict),
+            TELEMETRY_TYPE_DISTRIBUTION: defaultdict(dict),
+        }  # type: Dict[str, Dict[str, Dict]]
 
     def get(self):
         # type: () -> Dict[str, Any]
         namespace_metrics = self._metrics_data.copy()
```

Comment:
oof this copy is probably fairly heavy, no ?

.... do we ever remove/reset the existing points on a metric?

it seems like gauge and rate metrics will function ok. count metrics will never update it's timestamp, so we will always send the first timestamp and an absolute count ever growing? and distribution metrics are a memory leak?

also distribution is missing type, common, and timestamp in `to_dict`?

it feels like a missed opportunity to have something like:

```python
def reset(self):
    points, self._points = self._points, []
    return points
```

Reply:
Yup, I am not benchmarking this. I focused on just optimizing queuing metrics because that is a synchronous operation. I plan to get to this next.